consensus/src/sync/syncer.rs

```diff
@@ -248,47 +249,15 @@ impl<N: Network, M: MacroSync<N::PeerId>, L: LiveSync<N>> Syncer<N, M, L> {
             }
         };
 
-        // Fetch the reported election block from our chain.
-        // If we don't know the block, the peer is either ahead of us or on a different chain.
-        // In this case, we conservatively assume that the peer is not synced.
-        let blockchain = blockchain.read();
-        let election_block = match blockchain.get_block(&head.election, false) {
-            Ok(block) => block,
-            Err(_) => {
-                debug!(
-                    %peer_id,
-                    election_head = %head.election,
-                    "Incompatible peer's election block not found"
-                );
-                return false;
-            }
-        };
-
-        // If the peer is in a different epoch than us, we assume that it's not synced.
-        if election_block.epoch_number() != blockchain.election_head().epoch_number() {
-            debug!(
-                %peer_id,
-                peers_epoch = election_block.epoch_number(),
-                our_epoch = blockchain.election_head().epoch_number(),
-                "Incompatible peer is in a different epoch"
-            );
-            return false;
-        }
-
-        // Fetch the reported checkpoint block from our chain.
-        // If we don't know the block, the peer is either ahead of us or on a different chain,
-        // We consider the peer as not in sync until we catch up and is sufficiently close (see below).
-        let checkpoint = match blockchain.get_block(&head.r#macro, false) {
-            Ok(block) => block,
-            Err(_) => return false,
-        };
-
-        // We consider the peer synced if it's at most one batch behind us.
-        if checkpoint.batch_number() < blockchain.macro_head().batch_number().saturating_sub(1) {
```

Comment:
This does consider the peer synced as soon as they are in the same batch or the one before (I made another comment about it). As a checkpoint is final by the security assumption this should not be a real problem.

The peers head state is however not in any way verifiable. Previously it could be sanity checked that at least the macro chain is the same, now they can just claim to be at whatever height, making up a hash. That may or may not be problematic, I am not too familiar with the caller side here.

Reply:
It's not a problem if the peer is on a different chain _within the same batch_. That other chain might become the canonical one after all, so we want to keep accepting blocks from that peer.

The function name `is_peer_synced` is probably a bit misleading. `is_peer_macro_synced` or `is_peer_eligible_for_live_sync` would better capture the semantics of this function.

The peer's claimed state is never verifiable, irrespective of whether it contains a hash of their state, so we're not losing anything in terms of verifiability here. However, we could consider reintroducing the checkpoint hash to check if the security assumption has been violated and ban peers that are on a different macro chain.

Reply:
As it is just an additional hash which does provide some utility I would suggest to have it remain in the response. It is not the most solid "proof" of a peers local state if equal but if different it is an interesting observation to make.

Reply:
The renaming is simple and a good idea. I think it's ok to leave the rest as is